### PR TITLE
feat(volo-http): support `ConnectionInfo`

### PR DESCRIPTION
**Pull request by yukiiiteru**
## Motivation

This PR implemented `ConnectionInfo` with reference to `actix-web`.

### CODE REVIEW

File: volo-http/src/info.rs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+///
+/// If the older, related headers are also present (eg. `X-Forwarded-For`), then `Forwarded`
+/// is preferred.
+///
+/// [rfc7239]: https://datatracker.ietf.org/doc/html/rfc7239
+/// [rfc7239-62]: https://datatracker.ietf.org/doc/html/rfc7239#section-6.2
+/// [rfc7239-63]: https://datatracker.ietf.org/doc/html/rfc7239#section-6.3
+#[derive(Debug, Clone)]
+pub struct ConnectionInfo {
+    host: String,
```
[the remark this answers is not in the document]
Comment (wfly1998) [reply volo-http/src/info.rs]: `url::Host` supports only host name without port, but the host may be `foo.bar.com:8080` or `127.0.0.1:8080`, so it does not work here.

File: volo-http/src/info.rs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    scheme: String,
+    peer_addr: String,
```
[the remark this answers is not in the document]
Comment (wfly1998) [reply volo-http/src/info.rs]: In fact there is a field `volo::net::Address` in `HttpContext`, maybe we should copy it again?

File: volo-http/src/info.rs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+/// If the older, related headers are also present (eg. `X-Forwarded-For`), then `Forwarded`
+/// is preferred.
+///
+/// [rfc7239]: https://datatracker.ietf.org/doc/html/rfc7239
+/// [rfc7239-62]: https://datatracker.ietf.org/doc/html/rfc7239#section-6.2
+/// [rfc7239-63]: https://datatracker.ietf.org/doc/html/rfc7239#section-6.3
+#[derive(Debug, Clone)]
+pub struct ConnectionInfo {
+    host: String,
+    scheme: String,
```
[the remark this answers is not in the document]
Comment (wfly1998) [reply volo-http/src/info.rs]: Done

File: volo-http/src/info.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (PureWhiteWu) [reply volo-http/src/info.rs]: Then maybe we should separate it into url::Host and a u16 port like url::Url does?

File: volo-http/src/info.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (PureWhiteWu) [reply volo-http/src/info.rs]: What is `peer_addr` for? Does it contains port information, or is it only for ip?

**Review by PureWhiteWu — APPROVED** [review]

**Pull request merged**